packages/core/src/Stores/ui-store.js

```diff
@@ -199,6 +202,12 @@ export default class UIStore extends BaseStore {
         }
     }
 
+    @action.bound
+    handlePrompt(when, cb = () => {}) {
```

Comment:
- Could we rename this to something like `setPromptHandler`?
- Could `when` be renamed to `condition`?